awx/api/views.py

```diff
@@ -2732,13 +2730,39 @@ def post(self, request, *args, **kwargs):
             if fd not in request.data and id_fd in request.data:
                 request.data[fd] = request.data[id_fd]
 
-        if get_request_version(self.request) == 1 and 'extra_credentials' in request.data:  # TODO: remove in 3.3
+        # This block causes `extra_credentials` to _always_ be ignored for
+        # the launch endpoint if we're accessing `/api/v1/`
+        if get_request_version(self.request) == 1 and 'extra_credentials' in request.data:
             if hasattr(request.data, '_mutable') and not request.data._mutable:
                 request.data._mutable = True
             extra_creds = request.data.pop('extra_credentials', None)
             if extra_creds is not None:
                 ignored_fields['extra_credentials'] = extra_creds
 
+        # Automatically convert legacy launch credential arguments into a list of `.credentials`
+        if 'credentials' not in request.data and (
+            'credential' in request.data or
+            'vault_credential' in request.data or
+            'extra_credentials' in request.data
+        ):
+            credentials = obj.credentials.all()
+            for key, conditional in (
+                ('credential', lambda cred: cred.credential_type.kind != 'ssh'),
+                ('vault_credential', lambda cred: cred.credential_type.kind != 'vault'),
+                ('extra_credentials', lambda cred: cred.credential_type.kind not in ('cloud', 'net'))
+            ):
+                if key in request.data:
+                    credentials = filter(conditional, credentials)
+                    prompted_value = request.data.pop(key)
+                    if not isinstance(prompted_value, Iterable):
+                        prompted_value = [prompted_value]
+                    credentials.extend(prompted_value)
+            if credentials:
+                request.data['credentials'] = map(
+                    lambda cred: cred if isinstance(cred, int) else cred.pk,
+                    credentials
+                )
+
```

Comment:
I don't see any guarantee that `request.data._mutable = True` will be triggered before hitting this code block, which would produce errors. If we don't need that anyway, then we could just delete it.

Reply:
Hrm.  Yea, I'm questioning why we have this now.  This PR definitely has tests that change `request.data` without falling into the `_mutable=True` block.